expr/sort.go

```diff
@@ -81,52 +81,54 @@ func NewSortFn(nullsMax bool, fields ...FieldExprResolver) SortFn {
 	}
 }
 
-func NewSortValFn(nullsMax bool) SortValFn {
+func NewKeyCompareFn(key *zng.Record) (KeyCompareFn, error) {
 	sorters := make(map[zng.Type]comparefn)
-	return func(a zng.Value, b zng.Value) int {
-		// Handle nulls according to nullsMax
-		nullA := isNull(a)
-		nullB := isNull(b)
-		if nullA && nullB {
-			return 0
+	it := key.FieldIter()
+	var accessors []FieldExprResolver
+	var keyval []zng.Value
+	for !it.Done() {
+		name, val, err := it.Next()
+		if err != nil {
+			return nil, err
 		}
-		if nullA {
-			if nullsMax {
-				return 1
-			} else {
+		// we got to unset... all remaining values inthe key
+		// must be unset
```

Comment:
```suggestion
		// once an unset value is seen, all remaining values are also unset
```

Reply:
I took Noah's suggestion; he beat you to it.